backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/InlineClasses.kt

```diff
@@ -96,7 +101,13 @@ enum class KonanPrimitiveType(val classId: ClassId, val binaryType: BinaryType.P
     val fqName: FqNameUnsafe get() = this.classId.asSingleFqName().toUnsafe()
 
     companion object {
-        val byFqName = KonanPrimitiveType.values().associateBy { it.fqName }
+        val byFqNameParts = KonanPrimitiveType.values().groupingBy { it.classId.packageFqName }
```

Comment:
The code below assumes that all classes are top-level. Please add a check to ensure that every class is top-level here.